### ISSUE TITLE
Clean migration methods

### ISSUE THREAD

**Issue by orthagh** [pluginsGLPI/formcreator#320]
I see lot of direct sql 'alter table' statements.
glpi standards needs to use Migration class to do this.

I started a clean for Form class in #316 but we other class are impacted.

**Comment by btry** [comment]
Should not this issue being invalidated by #453 and #480 ?

**Comment by orthagh** [comment]
Maybe, did you remove direct sql statements ?

**Comment by btry** [comment]
No, I did not.

I designed the rewrite with as least changes as possible compared to the original code. This code  (IMHO) should not evolve except for bugfixes. 

I'll replace most ALTER queries by calls to Migration class, but some of them seems to not have replacement (at least ENUM upgrade related queries).

About ENUM, I found the implementation in the plugin nice enough to consider evolutions of the Migration class. Such code would allow in the futire a complete cleanup of the legacy upgrade code of this plugin, benefit to others, and maybe GLPI if someday ENUMs are introduced.
